FlowCrypt/src/main/java/com/flowcrypt/email/security/pgp/PgpMsg.kt

```diff
@@ -329,21 +330,38 @@ object PgpMsg {
       else -> {
         try {
           val detectedRawBlocks = RawBlockParser.detectBlocks(part, isOpenPGPMimeSigned)
-          val attachmentRawBlocks =
-            detectedRawBlocks.filter { it.type == RawBlockParser.RawBlockType.ATTACHMENT }
+          val attachmentRawBlock =
+            detectedRawBlocks.firstOrNull { it.type == RawBlockParser.RawBlockType.ATTACHMENT }
+          val inlineAttachmentRawBlock =
+            detectedRawBlocks.firstOrNull { it.type == RawBlockParser.RawBlockType.INLINE_ATTACHMENT }
 
-          for (rawBlock in attachmentRawBlocks) {
+          attachmentRawBlock?.let {
             blocks.add(
               MsgBlockFactory.fromAttachment(
                 type = MsgBlock.Type.PLAIN_ATT,
-                attachment = part as MimePart,
+                rawBlock = it,
+                mimePart = part as MimePart,
+                isOpenPGPMimeSigned = isOpenPGPMimeSigned
+              )
+            )
+          }
+
+          inlineAttachmentRawBlock?.let {
+            blocks.add(
+              MsgBlockFactory.fromAttachment(
+                type = MsgBlock.Type.INLINE_PLAIN_ATT,
+                rawBlock = it,
+                mimePart = part as MimePart,
                 isOpenPGPMimeSigned = isOpenPGPMimeSigned
               )
             )
           }
 
           val msgBlocks = processRawBlocks(
-            rawBlocks = detectedRawBlocks - attachmentRawBlocks.toSet(),
+            rawBlocks = detectedRawBlocks.toMutableList().apply {
+              inlineAttachmentRawBlock?.let { remove(it) }
+              inlineAttachmentRawBlock?.let { remove(it) }
```

Comment:
Why did you call this twice?

Reply:
Thank you. It's a good one. Copy-paste issue.

Reply:
Should be

```
attachmentRawBlock?.let { remove(it) }
inlineAttachmentRawBlock?.let { remove(it) }
```

Reply:
Sounds good!